### PR TITLE
Code for Union Find in Java

### PR DESCRIPTION
**Pull request by Shubhkishore**

### CODE REVIEW

File: Algorithms/UnionFind.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+
+/**
+ * @author johnny16
+ *
+ */
```
Comment (vanshikaarora) [inline Algorithms/UnionFind.java]: Hey, could you please remove this comment

File: Misc/Pair.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+class IntIntPair implements Comparable<IntIntPair> {
```
Comment (vanshikaarora) [inline Misc/Pair.java]: Wasn't this file already merged [here](https://github.com/vanshikaarora/DS_Algo/pull/19/files#diff-fb953f72ba2705f9a4d1920911cd5128)

File: Algorithms/UnionFind.java (hunk 1 above)
Comment (Shubhkishore) [reply Algorithms/UnionFind.java]: Done

File: Misc/Pair.java (hunk 2 above)
Comment (Shubhkishore) [reply Misc/Pair.java]: Sorry, my bad.

File: Algorithms/UnionFind.java (hunk 1 above)
Comment (vanshikaarora) [reply Algorithms/UnionFind.java]: Have you pushed the commit with the recent changes?

**Comment by vanshikaarora** [comment]
Nice to the changes especially comments in the code.

I am happy to merge:)

**Comment by Shubhkishore** [comment]
Comments always help in code re-usability :).

Thanks!

**Pull request merged**